packages/next-app/src/contexts/PoolMetadataContext.tsx

```diff
@@ -0,0 +1,93 @@
+"use client";
+
+import cuid from "cuid";
+import { createContext, ReactNode, useState } from "react";
+
+// import metadataGql from "../lib/poolMetadataGql";
+
+const cellData: {
+  id: string;
+  name: string;
+  type: string;
+  desc: string;
+  value: React.ReactNode;
+}[] = [
+  {
+    id: cuid(),
+    name: "Pool Address",
+    type: "address",
+    desc: "The address of the smart contract that implements the exchange pool",
+    value: "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D",
+  },
+  {
+    id: cuid(),
+    name: "Pool link",
+    type: "URL",
+    desc: "The address of the smart contract that implements the exchange pool",
+    value: "https://github.com/",
+  },
+  {
+    id: cuid(),
+    name: "Pool image",
+    type: "image",
+    desc: "balancer logo",
+    value: "https://s2.coinmarketcap.com/static/img/coins/200x200/5728.png",
+  },
+];
+
+export interface PoolMetadataAttribute {
+  id: string;
+  name: string;
+  type: string;
+  desc: string;
+  value: unknown;
+}
+
+interface PoolMetadataContextType {
+  metadata: PoolMetadataAttribute[];
+  handleAddMetadata: (data: PoolMetadataAttribute) => void;
+  handleUpdateMetadata: (data: PoolMetadataAttribute) => void;
+  selectedPool: string | null;
+  handleSetPool: (poolId: string) => void;
+}
+
+export const PoolMetadataContext = createContext({} as PoolMetadataContextType);
+
+export function PoolMetadataProvider({ children }: { children: ReactNode }) {
+  const [selectedPool, setSelectedPool] = useState<string>("");
+  // const { data: poolMetadata } = metadataGql.useMetadataPool({
+  //   poolId: selectedPool,
+  // });
+  // cellData = getDataFromIpfs(poolMetadata.CID)
```

Comment:
is it intended to get current metadata if it exists? will be done in another issue?